drivers/clock_control/clock_control_nrf_auxpll.c

```diff
@@ -0,0 +1,175 @@
+		 (config->ref_clk_hz * (uint64_t)config->frequency) / 65536U) /
+		config->out_div;
+
+	return 0;
+}
+
+static enum clock_control_status clock_control_nrf_auxpll_get_status(const struct device *dev,
+								     clock_control_subsys_t sys)
+{
+	const struct clock_control_nrf_auxpll_config *config = dev->config;
+	uint32_t val;
+
+	ARG_UNUSED(sys);
+
+	val = sys_read32(config->base + AUXPLL_STATUS) & AUXPLL_STATUS_RUNNING_LOCKED;
+	if (val == AUXPLL_STATUS_RUNNING_LOCKED) {
+		return CLOCK_CONTROL_STATUS_ON;
+	}
+
+	return CLOCK_CONTROL_STATUS_OFF;
+}
+
+static struct clock_control_driver_api clock_control_nrf_auxpll_api = {
+	.on = clock_control_nrf_auxpll_on,
+	.off = clock_control_nrf_auxpll_off,
+	.get_rate = clock_control_nrf_auxpll_get_rate,
+	.get_status = clock_control_nrf_auxpll_get_status,
+};
+
+static int clock_control_nrf_auxpll_init(const struct device *dev)
+{
+	const struct clock_control_nrf_auxpll_config *config = dev->config;
+
+	sys_write32(config->frequency, config->base + AUXPLL_AUXPLLCTRL_FREQUENCY);
+
+	sys_write32(1U, config->base + AUXPLL_MIRROR);
+	sys_write32(sys_read32(config->ficr_ctune), config->base + AUXPLL_TRIM_CTUNE);
+	sys_write32(config->cfg, config->base + AUXPLL_CONFIG_CFGSTATIC);
+	sys_write32(config->out_div, config->base + AUXPLL_AUXPLLCTRL_OUTSEL);
+	sys_write32(0U, config->base + AUXPLL_MIRROR);
+
+	sys_write32(AUXPLL_AUXPLLCTRL_MODE_MODECTRL_LOCKED, config->base + AUXPLL_AUXPLLCTRL_MODE);
+
+	return 0;
+}
+
+#define CLOCK_CONTROL_NRF_AUXPLL_DEFINE(n)                                                         \
+	static const struct clock_control_nrf_auxpll_config config##n = {                          \
+		.base = DT_INST_REG_ADDR(n),                                                       \
+		.ref_clk_hz = DT_PROP(DT_INST_CLOCKS_CTLR(n), clock_frequency),                    \
+		.ficr_ctune = DT_REG_ADDR(DT_INST_PHANDLE(n, nordic_ficrs)) +                      \
+			      DT_INST_PHA(n, nordic_ficrs, offset),                                \
+		.cfg = (((DT_INST_PROP(n, nordic_out_drive)                                        \
+			  << AUXPLL_CONFIG_CFGSTATIC_OUTDRIVE_POS) &                               \
+			 AUXPLL_CONFIG_CFGSTATIC_OUTDRIVE_MSK) |                                   \
+			((DT_INST_PROP(n, nordic_current_tune)                                     \
+			  << AUXPLL_CONFIG_CFGSTATIC_SELCONSTANTI_POS) &                           \
+			 AUXPLL_CONFIG_CFGSTATIC_SELCONSTANTI_MSK) |                               \
+			(DT_INST_PROP(n, nordic_sdm_disable) * AUXPLL_CONFIG_CFGSTATIC_SDMOFF) |   \
+			(DT_INST_PROP(n, nordic_dither_disable) *                                  \
+			 AUXPLL_CONFIG_CFGSTATIC_SDMDITHEROFF) |                                   \
+			((DT_INST_ENUM_IDX(n, nordic_range)                                        \
+			  << AUXPLL_CONFIG_CFGSTATIC_AUXPLLRANGE_POS) &                            \
+			 AUXPLL_CONFIG_CFGSTATIC_AUXPLLRANGE_MSK)),                                \
```

Comment:
The `FIELD_PREP()` macro could perhaps make this a bit easier to read (and the `*_POS` symbols would no longer be needed then).